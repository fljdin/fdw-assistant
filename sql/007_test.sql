
INSERT INTO config (source, target, pkey, trunc) VALUES
-- a truncate should failed if target is involved in a foreign key
    ('source.parent', 'public.parent', 'id', true);

SELECT invocation FROM plan('{public.parent}');

-- copy(12) should fail because "parent" is referenced by "child"
CALL copy(12);
