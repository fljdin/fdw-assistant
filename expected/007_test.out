INSERT INTO config (source, target, pkey, trunc) VALUES
-- a truncate should failed if target is involved in a foreign key
    ('source.parent', 'public.parent', 'id', true);
INSERT 0 1
SELECT invocation FROM plan('{public.parent}');
     invocation     
--------------------
 CALL fdw.copy(12);
(1 row)

-- copy(12) should fail because "parent" is referenced by "child"
CALL copy(12);
NOTICE:  Executing: TRUNCATE public.parent
ERROR:  cannot truncate a table referenced in a foreign key constraint
CONTEXT:  SQL statement "TRUNCATE public.parent"
PL/pgSQL function copy(bigint) line 27 at EXECUTE
CONTEXT:  PL/pgSQL function copy(bigint) line 33 at RAISE
